pkg/engine/context/resolvers/informerCache.go

```diff
@@ -0,0 +1,92 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func getSelectorForCache() (labels.Selector, error) {
+	expectedLabels := "cache.kyverno.io/enabled"
+	selector := labels.Everything()
+	requirement, err := labels.NewRequirement(expectedLabels, selection.Exists, nil)
+	if err == nil {
+		selector = selector.Add(*requirement)
+	}
+	return selector, err
```

Comment:
We return a selector in case of error ?

Reply:
it's an empty selector. we are checking for `err` value

Reply:
if there is an error, we should return nil, err

Reply:
we should not even create a factory

Reply:
fixed